Sources/SotoCore/Credential/ConfigFileCredentialProvider.swift

```diff
@@ -43,22 +31,24 @@ class AWSConfigFileCredentialProvider: CredentialProviderSelector {
     /// internal version of internal provider. Should access this through `internalProvider`
     var _internalProvider: CredentialProvider?
 
-    init(credentialsFilePath: String, profile: String? = nil, context: CredentialProviderFactory.Context) {
+    init(credentialsFilePath: String, configFilePath: String? = nil, profile: String? = nil, context: CredentialProviderFactory.Context) {
         self.startupPromise = context.eventLoop.makePromise(of: CredentialProvider.self)
         self.startupPromise.futureResult.whenSuccess { result in
             self.internalProvider = result
```

Comment:
@adam-fowler question about this. Here we assign the result from `startupPromise` to `internalProvider`. Does this mean that when `CredentialProviderSelector.getCredential(...)` is called, `internalProvider` will always have a value?

I'm trying to figure out at what point should I set the value for `internalProvider` to use STS assume role operation.

Reply:
All you need to do is return the CredentialProvider from `getSharedCredentialsFromDisk`. This is then progated to _internalProvider via `self.startupPromise`. See line 37

Reply:
Sounds good, working on file loading now, should have an update later today.